sysend.js

```diff
@@ -504,6 +514,29 @@
                 sysend.emit('__close__', { id: target_id, wasPrimary: primary });
             }, { capture: true });
 
+            // Track when receiving messages from iframes
+            window.addEventListener("message", function(e) {
+                if (typeof e.data === 'string' && e.data.match(prefix_re)) {
+                    try {
+                        var payload = JSON.parse(e.data);
+                        if (!payload || payload.name !== uniq_prefix) return;
+
+                        var data = unserialize(payload.data);
+
+                        if (data.data && data.data.target && payload.iframe_id && payload.iframe_id == data.data.target) {
+                            data.data.target = target_id; // Swap target ID's since this call came from iframe
+                        }
+
+                        var key = data && data.name;
+                        if (callbacks[key])
+                            invoke(key, data.data);
```

Comment:
And please use curly braces on if statements. Maybe I should set up ESLint for this.